src/go/rpk/pkg/vcloud/yak/client.go

```diff
@@ -67,6 +69,40 @@ func (yc *YakClient) GetNamespaces() ([]*Namespace, error) {
 	return namespaces, nil
 }
 
+func (yc *YakClient) GetClusters(namespaceName string) ([]*Cluster, error) {
+	token, err := yc.conf.ReadToken()
+	if err != nil {
+		return nil, ErrLoginTokenMissing{err}
+	}
+	// map namespace name to namespace id
+	namespaceId, err := yc.getNamespaceId(namespaceName)
+	if err != nil {
+		return nil, err
+	}
+	// retrieve clusters
+	url := fmt.Sprintf("%s/%s", DefaultYakUrl, fmt.Sprintf(getClustersRouteTemplate, namespaceId))
+	resp, err := yc.get(token, url)
+	if err != nil {
+		return nil, fmt.Errorf("error calling api: %w", err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("error reading request body: %w", err)
+	}
+	err = yc.handleErrResponseCodes(resp, body)
```

Comment:
I think this common error handling logic should be moved to `get`, to avoid code duplication.

Reply:
yeah, but then get would have to read the body which complicates things a bit, I would have to return the body as bytes from get then. But could be done if you find it nicer

Reply:
done